LayoutTests/accessibility/mac/child-update-during-ax-request.html

```diff
@@ -0,0 +1,141 @@
+// because the accessibility tree becomes broken.
+
+setInterval(() => {
+    let randomString = "";
+    const characters = "ABCDEPXYZabcdef89";
+    while (randomString.length < 3)
+        randomString += characters.charAt(Math.floor(Math.random() * characters.length));
+
+    // Cause constant children changed events to ensure the accessibility tree is always processing updates.
+    document.getElementById("container").innerHTML = `<div>${randomString}</div>`;
+}, 0);
+
+var axTable = window.accessibilityController ? accessibilityController.accessibleElementById("table") : null;
+setInterval(() => {
+    if (window.accessibilityController) {
+        // `scrollToMakeVisible()` is intentionally used here, as it is processed asynchronously (it uses
+        // AccessibilityUIElement::executeOnAXThread and not executeOnAXThreadAndWait), making it possible
+        // for the accessibility thread to process queued tree updates (`AXIsolatedTree::applyPendingChanges`) while the
+        // main-thread is in the middle of queuing tree updates (`AXIsolatedTree::updateChildren`).
+        axTable.scrollToMakeVisible();
+    }
+}, 0);
+
+var root;
+var searchResult;
+var showTextContainer = true;
+var traversalCounter = 0;
+function traverse() {
+    traversalCounter++;
+    output += `TRAVERSAL ${traversalCounter}\n\n`;
+
+    let searchOutput = "";
+    let tableCount = 0;
+    let cellCount = 0;
+    let rowCount = 0;
+    let buttonCount = 0;
+    let foundThirdCellText = false;
+    let foundFourthCellText = false;
+
+    while (true) {
+        searchResult = root.uiElementForSearchPredicate(searchResult, true, "AXAnyTypeSearchKey", "", false);
+        if (!searchResult)
+            break;
+        
+        const role = searchResult.role;
+        const roleLowerCase = role.toLowerCase();
+        if (roleLowerCase.includes("table"))
+            tableCount++; 
+        else if (roleLowerCase.includes("cell"))
+            cellCount++;
+        else if (roleLowerCase.includes("row"))
+            rowCount++;
+        else if (roleLowerCase.includes("button"))
+            buttonCount++;
+
+        const id = searchResult.domIdentifier;
+        let resultDescription = `${id ? `#${id} ` : ""}${role}`;
+        if (role.includes("StaticText")) {
+            const textValue = ` ${accessibilityController.platformName === "ios" ? searchResult.description : searchResult.stringValue}`;
+            if (textValue.includes("ThirdCell"))
+                foundThirdCellText = true;
+            else if (textValue.includes("FourthCell"))
+                foundFourthCellText = true;
+
+            resultDescription += textValue;
+        }
+        searchOutput += `\n{${resultDescription}}\n`;
+    }
+
+    // Every traversal must include these elements. If not, it's very likely the accessibility tree is broken.
+    if (tableCount != 1 || cellCount != 4 || rowCount != 2 || buttonCount != 2 || !foundThirdCellText || !foundFourthCellText) {
```

Comment:
Could it just diff searchOutput relative to the initial rather than counting a bunch of things? Or does that make it harder to debug?

Reply:
This is a personal preference, but I find the failure logs that include counts + bools (like this) easier to diagnose, and a quick way to see why a test might be failing. Diffs don't make that immediately clear.

Reply:
> Could it just diff searchOutput relative to the initial rather than counting a bunch of things?
Normally I would do this, but because this test constantly adding random strings to the webpage, and hiding and un-hiding a div with text, the `searchOutput` diff could be different in non-important ways. I also figured this would be faster than doing a bunch of JavaScript string `includes` checks, as the output string is large, and this test is already a bit slow. But certainly not set in stone if you have another idea!